bumpversion/version_part.py

```diff
@@ -58,13 +76,206 @@ def __format__(self, format_spec):
         return self.value
 
     def __repr__(self):
-        return '<bumpversion.VersionPart:{}:{}>'.format(
-            self.config.__class__.__name__,
-            self.value
+        return "<bumpversion.VersionPart:{}:{}>".format(
+            self.config.__class__.__name__, self.value
         )
 
     def __eq__(self, other):
         return self.value == other.value
 
     def null(self):
         return VersionPart(self.config.first_value, self.config)
+
+
+class Version(object):
+    def __init__(self, values, original=None):
+        self._values = dict(values)
+        self.original = original
+
+    def __getitem__(self, key):
+        return self._values[key]
+
+    def __len__(self):
+        return len(self._values)
+
+    def __iter__(self):
+        return iter(self._values)
+
+    def __repr__(self):
+        return "<bumpversion.Version:{}>".format(keyvaluestring(self._values))
+
+    def bump(self, part_name, order):
+        bumped = False
+
+        new_values = {}
+
+        for label in order:
+            if label not in self._values:
+                continue
+            elif label == part_name:
+                new_values[label] = self._values[label].bump()
+                bumped = True
+            elif bumped:
+                new_values[label] = self._values[label].null()
+            else:
+                new_values[label] = self._values[label].copy()
+
+        new_version = Version(new_values)
+
+        return new_version
+
+
+class VersionConfig(object):
+
+    """
+    Holds a complete representation of a version string
+    """
+
+    def __init__(self, parse, serialize, search, replace, part_configs=None):
+
+        try:
+            self.parse_regex = re.compile(parse, re.VERBOSE)
+        except sre_constants.error as e:
```

Comment:
Probably also better to use `re.error` here

Reply:
Agree, will add a TODO. Again this is copied over so I'd rather address it in a subsequent PR.